core/domain/story_fetchers.py

```diff
@@ -321,6 +323,73 @@ def get_story_summaries_by_ids(
     return story_summaries
 
 
+def get_learner_group_syllabus_story_summaries(
+    story_ids: List[str]
+) -> List[story_domain.LearnerGroupSyllabusStorySummaryDict]:
+    """Returns the learner group syllabus story summary dicts
+    corresponding the given story ids.
+
+    Args:
+        story_ids: list(str). The list of story ids for which the story
+            summaries are to be returned.
+
+    Returns:
+        list(LearnerGroupSyllabusStorySummaryDict). The story summaries
+        corresponds to given story ids.
+    """
+    # If story check is applied only for mypy type checks. All story ids are
+    # supposed to be valid as a part of validation done for learner group
+    # syllabus before calling this function.
+    all_stories = [
+        story for story in get_stories_by_ids(story_ids) if story
+    ]
+
+    topic_ids = list(
+        {story.corresponding_topic_id for story in all_stories}
+    )
+    topics = topic_fetchers.get_topics_by_ids(topic_ids)
+    topic_id_to_topic_map = {}
+    for topic in topics:
+        # Ruling out the possibility of None for mypy type checking.
```

Comment:
This needs more details to substantiate why we use assert rather than raising exception. Please talk to @sahiljoster32, he can give you advice on this.

Reply:
Updated this comment with more reasoning. Thanks!